tripy/tripy/frontend/trace/ops/utils.py

```diff
@@ -49,24 +50,34 @@ def get_broadcast_dim(dim1, dim2):
 
 
 ##
-## Handling shape outputs: These are common policies to use for overring infer_shape_output_idxs
+## Handling returning different tensor variants (Shape or ShapeScalars) from operators
 ##
 
+"""
+Enum representing the tensor variants encoded in the result of `infer_tensor_variants`.
+"""
 
-class ShapeOutputIdxPolicies:
+
+class TensorVariants(Enum):
+    SHAPE = 1
+    SCALAR = 2
```

Comment:
We can actually just use the `Shape` and `ShapeScalar` types directly instead of this enum. With that and the other change I suggested, the call-site can be simplified to:
```py
# If there are any shape outputs, we infer the length. 
if any(typ == Shape for typ in res.value):
    inferred_lengths = op.infer_len()

for idx, typ in enumerate(res.value):
    if typ is not None:
        outputs[idx] = typ(outputs[idx])
        if inferred_lengths[idx] is not None:
            out_trace_tensors[idx].shape = [inferred_lengths[idx]]
```

Reply:
Not introducing the enum is a good idea too.